src/planner.cc

```diff
@@ -38,17 +47,77 @@ namespace {
 // The speed is initially the aimed goal; if it cannot be reached, the
 // AxisTarget will be modified to contain the actually reachable value. That is
 // used in planning along the path.
+// The starting_speed is the required speed at the start of the segment.
+// It is used to combine segments with different trajectory angles.
+// In the AxisTarget we define "motion invariants" (or targets). Values and
+// parameters that assuming the requested final trajectory stays the same, they
+// are constants, no matter the previous or next segments.
 struct AxisTarget {
   int position_steps[GCODE_NUM_AXES];  // Absolute position at end of segment.
-                                       // In steps.
+                                       // In steps. (steps)
 
   // Derived values
-  int delta_steps[GCODE_NUM_AXES];     // Difference to previous position.
+  int delta_steps[GCODE_NUM_AXES];  // Difference to previous position. (steps)
   enum GCodeParserAxis defining_axis;  // index into defining axis.
-  double speed;             // (desired) speed in steps/s on defining axis.
+  double start_speed;  // Starting speed of the segment (steps/s).
+  double speed;        // Maximum speed of the defining axis (steps/s).
+  double accel;        // Maximum acceleration of the defining axis (steps/s^2).
   unsigned short aux_bits;  // Auxillary bits in this segment; set with M42
-  double dx, dy, dz;        // 3D delta_steps in real units
-  double len;               // 3D length
+  double dx, dy, dz;        // 3D delta_steps in real units (mm)
+  double len;               // 3D length (mm)
+
+  std::string ToString() const {
+    std::ostringstream ss;
+    ss << "{";
+    ss << "\"start_speed\":" << start_speed << ",";
+    ss << "\"speed\":" << speed << ",";
+    ss << "\"accel\":" << accel << ",";
+    ss << "\"defining_axis\":" << defining_axis << ",";
+    ss << "\"delta\":"
+       << "[";
+    bool first = true;
+    for (const GCodeParserAxis a : AllAxes()) {
+      if (first) {
+        first = false;
+      } else {
+        ss << ",";
+      }
+      ss << delta_steps[a];
+    }
+    ss << "]}";
+    return ss.str();
+  }
+};
+
+// Define values that might change depending on new segments added to
+// the planning buffer.
+// Currently planned profile for an AxisTarget.
+//    |  v1
+//    |   /¯¯¯¯¯¯¯¯¯¯\
```

Comment:
to avoid having trouble with the backslas at the end of the line, maybe add a comment at end
```
//    |  v1
//    |   /¯¯¯¯¯¯¯¯¯¯\    // trapezoidal curve.
//    |  /            \v2
//    | /
// v0 |/________________
//      acc  travel   dec
```

Reply:
fixed by using `/* */`